Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -36,20 +36,39 @@ public final class CacheAdvance<T: Codable> {
     /// - Warning: `shouldOverwriteOldMessages` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will prevent appending new messages to this cache.
     /// - Warning: `decoder` must have a consistent implementation for the life of a cache. Changing this value after logs have been persisted to a cache may prevent reading messages from this cache.
     /// - Warning: `encoder` must have a consistent implementation for the life of a cache. Changing this value after logs have been persisted to a cache may prevent reading messages from this cache.
-    public init(
+    public convenience init(
         fileURL: URL,
         maximumBytes: Bytes,
         shouldOverwriteOldMessages: Bool,
         decoder: MessageDecoder = JSONDecoder(),
         encoder: MessageEncoder = JSONEncoder())
         throws
     {
-        self.fileURL = fileURL
-
-        writer = try FileHandle(forWritingTo: fileURL)
-        reader = try CacheReader(forReadingFrom: fileURL)
-        header = try CacheHeaderHandle(forReadingFrom: fileURL, maximumBytes: maximumBytes, overwritesOldMessages: shouldOverwriteOldMessages)
+        self.init(
+            fileURL: fileURL,
+            writer: try FileHandle(forWritingTo: fileURL),
+            reader: try CacheReader(forReadingFrom: fileURL),
+            header: try CacheHeaderHandle(
+                forReadingFrom: fileURL,
+                maximumBytes: maximumBytes,
+                overwritesOldMessages: shouldOverwriteOldMessages),
+            decoder: decoder,
+            encoder: encoder)
+    }
 
+    /// An internal initializer with no logic. Can be used to create pathological test cases.
+    required init(
```

Comment:
we need a way to inject these private properties so that we can mangle them prior to executing our test.